Source/Core/Core/HW/DSPHLE/UCodes/AXVoice.h

```diff
@@ -264,7 +261,7 @@ u32 ResampleAudio(std::function<s16(u32)> input_callback, s16* output, u32 count
 
       s64 samp = (t0 * c[0] + t1 * c[1] + t2 * c[2] + t3 * c[3]) >> 15;
 
-      output[i] = (s16)samp;
+      output[i] = (s16)std::clamp(samp, s64(-0x8000), s64(0x7FFF));
```

Comment:
Can't the [SaturingCast](https://github.com/dolphin-emu/dolphin/blob/02309bd0acf030f5f641d427e55b73d59455c4b1/Source/Core/Common/MathUtil.h#L36) helper function be used here?